File: BehaviorPattern/Observer/Golang/Concretes/Sunflower.go
package concretes

import (
	"v/Interfaces"
	"fmt"
)

type Sunflower struct {
	PlantAction string
}

// Constructor
func NewSunflower() interfaces.Observer {
	return &Sunflower{}
}

// Overise update
func (p *Sunflower) Update(action string) {
	p.PlantAction = action
	fmt.Println("Peashooter: Zombie action detected: " + action)
}

File: BehaviorPattern/Observer/Java/Concretes/Sunflower.java
package Concretes;

import Interfaces.Observer;

public class Sunflower implements Observer {
    private String plantAction;

    @Override
    public void update(String action) {
        this.plantAction = action;
        System.out.println("Sunflower: Zombie action detected: " + action);
    }
}
